pytket/tests/distribution_test.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2019-2022 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import Counter
+from typing import cast
+from numpy import isclose
+from pytket.utils import (
+    ProbabilityDistribution,
+    EmpiricalDistribution,
+    convex_combination,
+)
+import pytest
+
+
+def test_distribution() -> None:
+    pd = ProbabilityDistribution({"a": 1 / 3, "b": 2 / 3})
+    with pytest.raises(ValueError):
+        pd = ProbabilityDistribution({"a": 1 / 3, "b": 1, "c": -1 / 3})
+    pd1 = ProbabilityDistribution({"a": 1 / 3, "b": 2 / 3 - 1e-15, "c": 1e-15})
+    assert pd1.support == set(["a", "b"])
+    assert isclose(pd1["a"], 1 / 3)
+    assert pd1["c"] == 0.0
+    assert pd == pd1
+    with pytest.raises(ValueError):
+        pd2 = ProbabilityDistribution({"a": 2 / 3, "b": 4 / 3})
+    pd3 = ProbabilityDistribution({"a": 2 / 9, "b": 4 / 9, "c": 1 / 3})
+    pd4 = convex_combination([(pd, 1 / 3), (pd3, 2 / 3)])
+    assert pd4 == ProbabilityDistribution({"a": 7 / 27, "b": 14 / 27, "c": 2 / 9})
+
+    ed1 = EmpiricalDistribution(Counter({"a": 1, "b": 2}))
```

Comment:
maybe this line onwards should be a separate test?

Reply:
Done.